firmware/defmt-test/tests/ui/test-uses-invalid-type-annotation.rs

```diff
@@ -0,0 +1,10 @@
+fn main() {}
+
+#[defmt_test::tests]
```

Comment:
the accompanying .stderr file is empty so this test is failing.
but also CI seems to not be running the UI tests because it's currently green. could you update the xtask to run the new UI tests in defmt-test?

Reply:
Running ui tests with `trybuild` on Rust **stable** and **nightly** has a few issues on its own, therefore the snapshot tests are only run in a new `xtask` command on **stable** to be consistent.

Reply:
Also removed this test for this PR, because `trybuild` is not passing `--cfg test` to `rustc` when running tests, therefore skipping code checks in the resulting `#[cfg(test)]` block. Therefore this test file will succeed, even though the resulting code would fail when for example used as a test in the app template testsuite.